flink-learning-common/src/test/java/com/zhisheng/common/utils/DateUtilTests.java

```diff
@@ -0,0 +1,71 @@
+package com.zhisheng.common.utils;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static com.zhisheng.common.utils.DateUtil.*;
+import static org.junit.Assert.*;
+
+public class DateUtilTests {
+
+    @Test
+    public void testFormat() throws ParseException {
+        SimpleDateFormat myDate1 = new SimpleDateFormat("yyyy-MM-dd");
+        SimpleDateFormat myDate2 = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        SimpleDateFormat myDate3 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        SimpleDateFormat myDate4 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.0");
+
+        assertEquals(myDate3.parse(myDate3.format(new Date())).getTime(), format(new Date()));
+
+        assertEquals(myDate1.format(new Date()), format(new Date().getTime(), DateUtil.YYYY_MM_DD));
+        assertEquals(myDate1.format(new Date()), format(new Date(), DateTimeZone.getDefault(), DateUtil.YYYY_MM_DD));
+
+        long testDate1 = myDate1.parse(myDate1.format(new Date())).getTime();
+        long testDate2 = myDate2.parse(myDate1.format(new Date())).getTime();
+        long testDate3 = myDate3.parse(myDate1.format(new Date())).getTime();
+        long testDate4 = myDate4.parse(myDate1.format(new Date())).getTime();
+
+        assertEquals(testDate1, format(myDate1.format(new Date()), YYYY_MM_DD));
+        assertEquals(testDate2, format(myDate2.format(new Date()), YYYY_MM_DD_HH_MM));
+        assertEquals(testDate3, format(myDate3.format(new Date()), YYYY_MM_DD_HH_MM_SS));
+        assertEquals(testDate4, format(myDate4.format(new Date()), YYYY_MM_DD_HH_MM_SS_0));
```

Comment:
IMO this test is a bit confusing. I think you should use a fixed date (create a `Date` object with a specific date-time) and also maybe you should split the tests for different overloads of `format` into different methods.

Also, you could use a fixed date so you don't have to call `SimpleDateFormat` in the assertion. That way the tests will only break if there is a regression in this codebase. As it stands the test will fail if `SimpleDateFormat` changes.

Reply:
Done